Source/Core/DolphinQt2/WiiUpdate.cpp

```diff
@@ -96,7 +96,10 @@ static WiiUtils::UpdateResult ShowProgress(QWidget* parent, Callable function, A
   auto* cancel_button = new QPushButton(QObject::tr("&Cancel"), parent);
   dialog.setCancelButton(cancel_button);
   Common::Flag was_cancelled;
-  QObject::disconnect(&dialog, &QProgressDialog::canceled, &dialog, &QProgressDialog::cancel);
+  // Disconnect the default cancel signal. We don't know if Qt connected it using the old
+  // or the new signal slot syntax, so we need to use the old syntax with a wildcard,
+  // which allows us to get rid of all slots in all cases.
+  QObject::disconnect(&dialog, SIGNAL(canceled()), nullptr, nullptr);
```

Comment:
I'm not sure -- do you know if using the new syntax with a wildcard works when the connection was made using the old syntax *and* the new one? The [docs](https://wiki.qt.io/New_Signal_Slot_Syntax) aren't particularly clear about this:

> #### Disconnecting
> As you might expect, there are some changes in how connections can be terminated in Qt5, too. [...]
> * Symetric to the function pointer one
> Only works if you connected with the symmetric call, with function pointers (Or you can also use 0 for wild card) In particular, does not work with static function, functors or lambda functions.

I understood that as saying that the new, better way of disconnecting will only work if the new syntax was used for connecting, and that this new way can only be used with function pointers or wildcards.

On the other hand, the documentation says this about the old way:

> You can disconnect in the old way (using SIGNAL, SLOT) but only if
> * You connected using the old way, or
> * If you want to disconnect all the slots from a given signal using wild card character

I wasn't (and still am not) sure if using the new syntax is reliable enough and will work even after Qt updates, whereas we have that guarantee with the old syntax, which is why I used the latter here.

If you're sure that it *will* work, let me know and I'll change it to use the new syntax.